### ISSUE TITLE
Training loss and val loss decrease normally, but accuracy is always 0

### ISSUE THREAD

**Issue by yisongbetter** [meijieru/crnn.pytorch#21]
I use the icdar dataset to train crnn(pytorch), here is the problem:
Both training loss and test loss decrease normally, but the network can not predict anything!
Dose anybody encounter similar problem?
Here is an screenshot of my result:

Start val
-------------------------- =>                     , gt: Oleg
-------------------------- =>                     , gt: P
-------------------------- =>                     , gt: CASTLE
-------------------------- =>                     , gt: HIGH
-------------------------- =>                     , gt: GYM
-------------------------- =>                     , gt: FOR
-------------------------- =>                     , gt: COOLING
-------------------------- =>                     , gt: only
-------------------------- =>                     , gt: Refill
-------------------------- =>                     , gt: GRADUATE
Test loss: 19.685625, accuray: 0.000000
[7/25][60/60] Loss: 19.398106
[8/25][20/60] Loss: 19.721890
[8/25][40/60] Loss: 19.135641
Start val
-------------------------- =>                     , gt: Hypertension:
-------------------------- =>                     , gt: ntl:
-------------------------- =>                     , gt: ANCIENT
-------------------------- =>                     , gt: PanaSyncE70
-------------------------- =>                     , gt: Specialists
-------------------------- =>                     , gt: 2N
-------------------------- =>                     , gt: EU-funded
-------------------------- =>                     , gt: Dez.
-------------------------- =>                     , gt: Guidex
-------------------------- =>                     , gt: TAXI
Test loss: 19.580940, accuray: 0.000000
[8/25][60/60] Loss: 20.216869
[9/25][20/60] Loss: 18.244850
[9/25][40/60] Loss: 19.222029
Start val
-------------------------- =>                     , gt: Connells
-------------------------- =>                     , gt: 2010
-------------------------- =>                     , gt: &
-------------------------- =>                     , gt: GENTLEMEN
-------------------------- =>                     , gt: WELCOME
-------------------------- =>                     , gt: STUDIO
-------------------------- =>                     , gt: DOLLAR
-------------------------- =>                     , gt: SPENCER
-------------------------- =>                     , gt: Systems
-------------------------- =>                     , gt: NIGHT
Test loss: 19.106692, accuray: 0.000000
[9/25][60/60] Loss: 18.840774

**Comment by abhiML** [comment]
did you run it with adadelta?

**Comment by yisongbetter** [comment]
When I use the pretrained model @meijieru offers, the problem gone, so maybe I have not get enough data to train my model from scratch.

**Comment by yisongbetter** [comment]
I have solve the problem,adadelta or adam is ok @abhiML.  I use 800 pictures to train crnn from scratch, which is stupid , when I use the retrained model, the problem disappear. 
Here is what I got:
<img width="1280" alt="screen shot 2017-05-18 at 01 17 11" src="https://cloud.githubusercontent.com/assets/17310875/26166938/dcc65daa-3b67-11e7-8eff-97f113d2bc38.png">

**Comment by abhiML** [comment]
Well I was using adam on the Synth dataset. It showed me infinite loss. But when I used adadelta it ran perfect. I did not use the pretrained model.

**Comment by yisongbetter** [comment]
@abhiML It should be the learning rate problem that bothers you, because adadelta requires no manual setting of a learning rate, so when using some momentum methods like adam, you should carefully set the initial learning rate, 1e-3 works well in my situation.

**Comment by meijieru** [comment]
Since it has been solved, I closed this issue temporally. Feel free to re-open it when necessary.

**Comment by SenWang-NEU** [comment]
I also encountered the same situation, because my data set was self-built. Is there any other way besides loading the pre-trained model?
